nodejs/awsx/apigateway/requestValidator.ts

```diff
@@ -0,0 +1,45 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// These APIs are currently experimental and may change.
+
+export type RequestValidator = "ALL" | "PARAMS_ONLY" | "BODY_ONLY";
+
+export interface Validators {
+    [name: string]: {
+        validateRequestBody: boolean;
+        validateRequestParameters: boolean;
+    };
+}
+
+export interface Parameter {
+    name: string;
+    in: Location;
+}
```

Comment:
these coudl all use documentation.

Reply:
for example, if i say `name: "foo", in: "path"`  does that mean a substring check that the path contains "foo"?  Same with "header". Does it mean the header just contains that string, or that there's a specific key in the header with that exact name.